### PR TITLE
expose jenkins client plugin e2e as ci-operator step

### PR DESCRIPTION
**Pull request by gabemontero**
duplicate what we did for jenkins sync plugin in https://github.com/openshift/release/pull/21474 
for the jenkins client plugin

### CODE REVIEW

**Review by adambkaplan — APPROVED** [review]
/lgtm

File: core-services/image-mirroring/openshift/mapping_origin_4_10 [hunk 1]
```diff
@@ -140,6 +140,7 @@ registry.ci.openshift.org/origin/4.10:jenkins quay.io/openshift/origin-jenkins:4
 registry.ci.openshift.org/origin/4.10:jenkins-agent-base quay.io/openshift/origin-jenkins-agent-base:4.10 quay.io/openshift/origin-jenkins-agent-base:4.10.0 quay.io/openshift/origin-jenkins-agent-base:latest
 registry.ci.openshift.org/origin/4.10:jenkins-agent-maven quay.io/openshift/origin-jenkins-agent-maven:4.10 quay.io/openshift/origin-jenkins-agent-maven:4.10.0 quay.io/openshift/origin-jenkins-agent-maven:latest
 registry.ci.openshift.org/origin/4.10:jenkins-agent-nodejs quay.io/openshift/origin-jenkins-agent-nodejs:4.10 quay.io/openshift/origin-jenkins-agent-nodejs:4.10.0 quay.io/openshift/origin-jenkins-agent-nodejs:latest
+registry.ci.openshift.org/origin/4.10:jenkins-client-plugin-e2e quay.io/openshift/origin-jenkins-client-plugin-e2e:4.10 quay.io/openshift/origin-jenkins-client-plugin-e2e:4.10.0 quay.io/openshift/origin-jenkins-client-plugin-e2e:latest
```
[the remark this answers is not in the document]
Comment (smarterclayton) [reply core-services/image-mirroring/openshift/mapping_origin_4_10:143]: I would prefer this be called `jenkins-client-plugin-tests` unless the `-e2e` part is deeply baked in to it.  Generally everything that tests in our repo ends with `-tests`.   If this is not net-new or a recent image, that's also a reason not to need to change.

[the remark this answers is not in the document]
Comment (gabemontero) [reply core-services/image-mirroring/openshift/mapping_origin_4_10]: image name updated to `jenkins-client-plugin-tests` @smarterclayton - thanks

[the remark this answers is not in the document]
Comment (gabemontero) [reply core-services/image-mirroring/openshift/mapping_origin_4_10]: thought I need to rename a file too evidently ...

**Comment by gabemontero** [comment]
jenkins got an out of mem error with the master branch run of our old, soon to be nuked, e2e-aws-jenkins @adambkaplan 

`java.lang.OutOfMemoryError: unable to create native thread: possibly out of memory or process/resource limits reached`

the 4.10 run passed

and more importantly, our new e2e's passed

I believe this is ready for lgtm, and moving on to just using the new e2e's in my next PR

**Review by adambkaplan — APPROVED** [review]
/lgtm

**Pull request merged**